### PR TITLE
pass options from CompositeView to detail views, since they may need …

### LINKED ISSUE #223: CompositeView needs to pass additional options to its components, i.e. detail views
the problem is that detail views can get additional options if they need them (such as an AppState if they need to operate on the state, etc.). also the CompositeView gets the additional options, but it needs to pass them to the detail views, because they need the options in reality, not the CompositeView itself

see also https://github.com/datavisyn/tdp_core/blob/master/src/extensions.ts#L307

### PR DESCRIPTION
**Pull request by lehnerchristian**
…them

closes #223

### CODE REVIEW

**Comment by thinkh** [comment]
@lehnerchristian Is there a related issue for this PR?

**Comment by lehnerchristian** [comment]
> @lehnerchristian Is there a related issue for this PR?

nope, this issue came up on the fly yesterday. the problem is that detail views can get additional options if they need them (such as an AppState if they need to operate on the state). also the CompositeView gets the additional options, but it needs to pass them to the detail views, because they need the options in reality, not the CompositeView itselt.

see also https://github.com/datavisyn/tdp_core/blob/master/src/extensions.ts#L307

I'll create a ticket with this text and link it

**Comment by lehnerchristian** [comment]
updated the description

File: src/views/CompositeView.ts [hunk 1]
```diff
@@ -469,9 +469,10 @@ export default class CompositeView extends EventHandler implements IView {
     const desc = (<ICompositeViewPluginDesc>this.context.desc);
 
     const toEntry = (desc: IElementDesc): Promise<ICompositeInfo> => {
+      const descOptions = desc.options || {};
       return Promise.resolve(desc.loader()).then((instance) => (<ICompositeInfo>{
         key: desc.key,
-        options: desc.options,
+        options: Object.assign({}, descOptions, this.options), // also pass the view options from the ViewWrapper to all views
```
Comment (dg-datavisyn) [inline src/views/CompositeView.ts]: Is this downward compatible?

**Pull request merged**